sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteRecordsInconsistent.java

```diff
@@ -0,0 +1,265 @@
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Maps;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A transform to write sharded records to BigQuery using the Storage API. This transform uses the
+ * default stream to write the records. Records written will show up in BigQuery immediately,
+ * however exactly once is not guaranteed - duplicates may appear in the output. For exactly-once
+ * writes, use {@link StorageApiWritesShardedRecords} or {@link StorageApiWriteUnshardedRecords}.
+ */
+@SuppressWarnings("FutureReturnValueIgnored")
+public class StorageApiWriteRecordsInconsistent<DestinationT, ElementT>
+    extends PTransform<PCollection<KV<DestinationT, byte[]>>, PCollection<Void>> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(StorageApiWriteRecordsInconsistent.class);
+
+  private final StorageApiDynamicDestinations<ElementT, DestinationT> dynamicDestinations;
+  private final CreateDisposition createDisposition;
+  private final String kmsKey;
+  private final BigQueryServices bqServices;
+  private final Coder<DestinationT> destinationCoder;
+  private static final ExecutorService closeWriterExecutor = Executors.newCachedThreadPool();
+
+  private static final Cache<String, StreamAppendClient> APPEND_CLIENTS =
+      CacheBuilder.newBuilder()
+          .expireAfterAccess(5, TimeUnit.MINUTES)
+          .removalListener(
+              (RemovalNotification<String, StreamAppendClient> removal) -> {
+                @Nullable final StreamAppendClient streamAppendClient = removal.getValue();
+                // Close the writer in a different thread so as not to block the main one.
+                runAsyncIgnoreFailure(closeWriterExecutor, streamAppendClient::close);
+              })
+          .build();
+
+  // Run a closure asynchronously, ignoring failures.
+  private interface ThrowingRunnable {
+    void run() throws Exception;
+  }
+
+  private static void runAsyncIgnoreFailure(ExecutorService executor, ThrowingRunnable task) {
+    executor.submit(
+        () -> {
+          try {
+            task.run();
+          } catch (Exception e) {
+            //
+          }
+        });
+  }
+
+  public StorageApiWriteRecordsInconsistent(
+      StorageApiDynamicDestinations<ElementT, DestinationT> dynamicDestinations,
+      CreateDisposition createDisposition,
+      String kmsKey,
+      BigQueryServices bqServices,
+      Coder<DestinationT> destinationCoder) {
+    this.dynamicDestinations = dynamicDestinations;
+    this.createDisposition = createDisposition;
+    this.kmsKey = kmsKey;
+    this.bqServices = bqServices;
+    this.destinationCoder = destinationCoder;
+  }
+
+  @Override
+  public PCollection<Void> expand(PCollection<KV<DestinationT, byte[]>> input) {
+    String operationName = input.getName() + "/" + getName();
+    // Append records to the Storage API streams.
+    return input.apply(
```

Comment:
Does it make sense to add `GroupIntoBatches.withShardedKey` in this transform? At least for streaming pipelines?

Reply:
Short answer: not really. The whole point of this transform is to avoid a shuffle at the cost of exactly once, letting the user trade off cost v.s. consistency. 

Long answer: there are some cases where that would make sense. This transform inherits the parallelism that is earlier in the graph, so if, e.g., the previous transform had a parallelism of 1 then so would this. Ideally we could detect this and automatically call GIB if that is the case. This sounds like future work though.

Reply:
gotcha. Thanks.